### PR TITLE
Reworked Legion feat. vectors and lazy sprites.

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

The legion megafauna has been reworked.

It now has three sizes but will split into three smaller skulls. The way splitting works has been changed to accommodate this change.

Total max HP is WAY down.

It has a new attack that spawn stationary turrets. The amount scales with skull size.

The charge attack now throws the Legion at the player. This deals some damage on impact and has audio clues.

Droprates for chests are now higher because of lower amounts of total skulls.

The sprites no longer use resize, thank fuck. Instead every size has its own sprite. The biggest one is only around 40% as large as it used to be. That's a good thing because the old sprite was WAY too big for its own good.

## Why It's Good For The Game

This hopefully results in a much more fun and dynamic fight against this boss. It used to be really slow and boring to fight legion for not much reward. Now you should get around the same level of reward for a faster but also harder fight.

https://www.youtube.com/watch?v=tFMJxqTIaYc&feature=youtu.be

As you can see here, the legion is much more proactive in melee because of the throw that has been added to the charge and it will usually block of areas of combat that have already been used with the turrets, meaning that you have to be constantly on the move.

It is intended that you need at least *some* form of AOE to beat this boss reliably in a reasonable amount of time.

## Changelog
:cl:
add: The legion megafauna has been reworked. The fight should now be both slightly harder and faster.
/:cl:

### CODE REVIEW

**Comment by wesoda25** [comment]
Much better than what it formerly was

**Comment by Shadowflame909** [comment]
Surprised to hear me saying this. But r u sure this makes it strong enough?

Legions have never been scarier then bubblegum or the colossus. Since he's just a floating skull. But it's still missing something.

Just nothing to really shake up the player. Maybe if that throw attack threw you? Like the wizard's repulse spell? Since its a skull tackling you.

**Comment by remove32** [comment]
making legion more dynamic than a bunch of skulls that facetank you is good for gameplay. i think if you add one or two other things to expand on it it'll be in a great place. this is already good but it could be better maybe

**Comment by Jungling** [comment]
Maybe since the legion gets a turret maybe the staff should summon a turret now. Or give you a charge attack like the stand. As someone who actively enjoys killing each of the megafuna legion has the worse rewards and if this change goes through all itll do is make the fight harder for the same reward. The main reason its never killed is just the lack of justification for the time it takes. its like a 8-15minute fight and the rewards don't justify the time when you could've killed 2 bubblegums or 2 ashe drakes instead.

**Comment by ExcessiveUseOfCobblestone** [comment]
I don't think the PR is meant to fix the issue of making players want to fight it for the purpose of the rewards.

**Comment by nemvar** [comment]
Yeah, having good gameplay is more important than having a good reward structure. We aren't some mobile game.

File: code/game/objects/structures/lavaland/legionturret.dm [hunk 1]
```diff
@@ -0,0 +1,90 @@
+///A basic turret that shoots at nearby mobs. Intended to be used for the legion megafauna.
+/obj/structure/legionturret
```
Comment (AnturK) [inline code/game/objects/structures/lavaland/legionturret.dm]: I really don't like this. Turrets are perf heavy and copy pasting version that's intended to be spammable is horrible idea.

Comment (nemvar) [reply code/game/objects/structures/lavaland/legionturret.dm]: Alright, I removed them from processing. They now only shoot once and then get deleted.

**Pull request merged**